packages/ra-core/src/auth/useLogoutIfAccessDenied.ts

```diff
@@ -76,11 +91,20 @@ const useLogoutIfAccessDenied = (): LogoutIfAccessDenied => {
                             : error && error.redirectTo
                             ? error.redirectTo
                             : undefined;
-                    logout({}, redirectTo);
+
+                    if (logoutUser) {
+                        logout({}, redirectTo);
+                    } else {
+                        const newLocation: LocationDescriptorObject = {
```

Comment:
Why do you create a new object? This forbids developers to pass a location object themselves. 

I'd pass the `redirectTo` directly, like so:

```
history.push(redirectTo);
```

Reply:
How do you recommend we do this then?

Specify the `redirectTo` object be a type of `LocationDescriptorObject`, and invert control up to the authProvider?

Reply:
Ignore the above comment, i had some issue with testing previously. I've pushed your suggestions and it works fine. 

I've tested using a regular path like `/unauthorized` and also adding query params. Are there any other manual test cases you recommend I try?